codebase/src/sorting/algorithms/selection.ts

```diff
@@ -18,51 +18,48 @@ export const selectionSort = async ({
   updateCurrentLines: (lines: number[]) => void;
   updateSortedIndex: (index: number | null) => void;
```

Comment:
There are lot of params here that only serve to update the UI. I would consider refactoring this API to make it more clear, ie `selectionSort(arr: Array, uiControls: { setHighlightedLines, ... })`

This makes the API easier to understand